stats.go

```diff
@@ -79,17 +80,30 @@ func (c *counter) report(name string, tags map[string]string, r StatsReporter) {
 	r.ReportCounter(name, tags, curr-prev)
 }
 
+func (c *counter) cachedReport() {
+	curr := atomic.LoadInt64(&c.curr)
+
+	prev := atomic.LoadInt64(&c.prev)
+	if prev == curr {
+		return
+	}
+	atomic.StoreInt64(&c.prev, curr)
```

Comment:
Might be best to avoid repeating this code by adding a new method that takes the value?

```go
func (c *counter) value() int64 {
	curr := atomic.LoadInt64(&c.curr)

	prev := atomic.LoadInt64(&c.prev)
	if prev == curr {
		return 0
	}
	atomic.StoreInt64(&c.prev, curr)
    return curr - prev
}
```